src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -4,109 +4,347 @@
+            }
+            fullParameterList.AddRange(parameters.Select(CreatePublicConstructorParameter));
+
             var summary = $"Initializes a new instance of {name}";
             var accessibility = usage == InputModelTypeUsage.Output
                 ? MethodSignatureModifiers.Internal
                 : MethodSignatureModifiers.Public;
 
-            return new ConstructorSignature(name, summary, null, accessibility, parameters.Select(CreatePublicConstructorParameter).ToList());
+            return new ConstructorSignature(
+                name,
+                summary,
+                null,
+                accessibility,
+                fullParameterList,
+                new(true, GetBaseObjectType()?.InitializationConstructor.Signature.Parameters ?? Array.Empty<Parameter>()));
         }
 
         private static Parameter CreatePublicConstructorParameter(Parameter p)
             => TypeFactory.IsList(p.Type) ? p with { Type = new CSharpType(typeof(IEnumerable<>), p.Type.IsNullable, p.Type.Arguments) } : p;
 
         private static Parameter CreateSerializationConstructorParameter(Parameter p) // we don't validate parameters for serialization constructor
             => p with { Validation = ValidationType.None };
+
+        protected override ObjectTypeConstructor BuildInitializationConstructor()
+        {
+            ObjectTypeConstructor? baseCtor = GetBaseObjectType()?.InitializationConstructor;
+
+            return new ObjectTypeConstructor(InitializationConstructorSignature, GetPropertyInitializers(Fields.PublicConstructorParameters), baseCtor);
+        }
+
+        protected override ObjectTypeConstructor BuildSerializationConstructor()
+        {
+            ObjectTypeConstructor? baseCtor = GetBaseObjectType()?.SerializationConstructor;
+
+            return new ObjectTypeConstructor(SerializationConstructorSignature, GetPropertyInitializers(Fields.SerializationParameters), baseCtor);
+        }
+
+        private ObjectPropertyInitializer[] GetPropertyInitializers(IReadOnlyList<Parameter> parameters)
+        {
+            List<ObjectPropertyInitializer> defaultCtorInitializers = new List<ObjectPropertyInitializer>();
+
+            Dictionary<string, Parameter> parameterMap = parameters.ToDictionary(
+                parameter => parameter.Name,
+                parameter => parameter);
+
+            foreach (var property in Properties)
+            {
+                // Only required properties that are not discriminators go into default ctor
+                // TODO map discriminator into cadl.json
+                //if (property == Discriminator?.Property)
+                //{
+                //    continue;
+                //}
+
+                ReferenceOrConstant? initializationValue;
+                Constant? defaultInitializationValue = null;
+
+                var propertyType = property.Declaration.Type;
+                if (property.SchemaProperty?.Schema is ConstantSchema constantSchema)
```

Comment:
Should we be using the models in the `Input` namespace here? Because I believe `ConstantSchema` is a class from modeler v4's model. Or this is a placeholder for future improvements?

Reply:
I will add a TODO here, this code is most likely a copy / paste and is no-op in its current state.

Reply:
Therefore the general principal is that we should never use Schema types from modeler v4 anymore once we are out of the constructor? We have a lot to refactor in mgmt generator - they coupled so deeply currently